{"url": "https://api.github.com/repos/mozilla/node-firefox/issues/20", "repository_url": "https://api.github.com/repos/mozilla/node-firefox", "labels_url": "https://api.github.com/repos/mozilla/node-firefox/issues/20/labels{/name}", "comments_url": "https://api.github.com/repos/mozilla/node-firefox/issues/20/comments", "events_url": "https://api.github.com/repos/mozilla/node-firefox/issues/20/events", "html_url": "https://github.com/mozilla/node-firefox/issues/20", "id": 56058092, "number": 20, "title": "Better error reporting / handling", "user": {"login": "sole", "id": 5609, "avatar_url": "https://avatars2.githubusercontent.com/u/5609?v=4", "gravatar_id": "", "url": "https://api.github.com/users/sole", "html_url": "https://github.com/sole", "followers_url": "https://api.github.com/users/sole/followers", "following_url": "https://api.github.com/users/sole/following{/other_user}", "gists_url": "https://api.github.com/users/sole/gists{/gist_id}", "starred_url": "https://api.github.com/users/sole/starred{/owner}{/repo}", "subscriptions_url": "https://api.github.com/users/sole/subscriptions", "organizations_url": "https://api.github.com/users/sole/orgs", "repos_url": "https://api.github.com/users/sole/repos", "events_url": "https://api.github.com/users/sole/events{/privacy}", "received_events_url": "https://api.github.com/users/sole/received_events", "type": "User", "site_admin": false}, "labels": [], "state": "open", "locked": false, "assignee": null, "assignees": [], "milestone": null, "comments": 4, "created_at": "2015-01-30T17:01:22Z", "updated_at": "2015-01-30T17:48:34Z", "closed_at": null, "author_association": "MEMBER", "body_html": "<p>While Promises are cool etc they also capture errors in a sneaky way and sometimes it's impossible to find where your error is unless you delve into the code and start console.logging like silly.</p>\n<p>For example I was passing a wrong parameter to <code>connect</code> and the promise was capturing the error in getWebAppsActor call, which overall remained silent. We should review the flow of each function and make sure errors can bubble up (perhaps intentionally building errors!)</p>\n<p>This is probably due to my own lack of experience with promises so I accept all fault already :-)</p>", "body_text": "While Promises are cool etc they also capture errors in a sneaky way and sometimes it's impossible to find where your error is unless you delve into the code and start console.logging like silly.\nFor example I was passing a wrong parameter to connect and the promise was capturing the error in getWebAppsActor call, which overall remained silent. We should review the flow of each function and make sure errors can bubble up (perhaps intentionally building errors!)\nThis is probably due to my own lack of experience with promises so I accept all fault already :-)", "body": "While Promises are cool etc they also capture errors in a sneaky way and sometimes it's impossible to find where your error is unless you delve into the code and start console.logging like silly. \n\nFor example I was passing a wrong parameter to `connect` and the promise was capturing the error in getWebAppsActor call, which overall remained silent. We should review the flow of each function and make sure errors can bubble up (perhaps intentionally building errors!)\n\nThis is probably due to my own lack of experience with promises so I accept all fault already :-)\n"}